bin/functions.js

```diff
@@ -0,0 +1,366 @@
+  checkShellResponse(options, spinner, res);
+  if (options.upstream + "/" + options.targetBranch != trim(res.stdout)) {
+    spinner.fail(`Target branch '${options.targetBranch}' does not have upstream '${options.upstream}'.`);
+    dryRunOrNoDryRun(options, () => shell.exit(1));
+  }
+  spinner.succeed(`Target branch '${options.targetBranch}' has upstream '${options.upstream}'.`);
+};
+
+const fastForwardAll = (options) => {
+  let spinner = ora(`Synchronizing branches...`).start();
+  fastforward(options, spinner, options.sourceBranch);
+
+  if (options.sourceBranch !== options.targetBranch) {
+    fastforward(options, spinner, options.targetBranch);
+    checkout(options, spinner, options.sourceBranch);
+  }
+  spinner.succeed("Branches synchronized");
+};
+
+const checkout = (options, spinner, branch) => {
+  if (spinner) {
+    spinner.color = "blue";
+    spinner.text = `Checking out ${branch}...`;
+  }
+  checkShellResponse(options, spinner, shell.exec(`git checkout ${branch} -q`));
+};
+
+const fastforward = (options, spinner, branch) => {
+  if (spinner) {
+    spinner.color = "blue";
+    spinner.text = `Fast-forwarding ${branch}...`;
+  }
+  checkShellResponse(options, spinner, shell.exec(`git checkout ${branch} -q`));
+
+  let res = shell.exec(`git rev-list "refs/heads/${branch}..refs/remotes/${options.upstream}/${branch}"`);
+  checkShellResponse(options, spinner, res);
+  if (trim(res.stdout) !== "") {
+    spinner.fail(`${branch} has local commits; can't fast-forward`);
+    dryRunOrNoDryRun(options, () => shell.exit(1));
+  }
+
+  res = shell.exec(`git merge --ff-only ${options.upstream}/${branch}`);
+  checkShellResponse(options, spinner, res);
+};
+
+const uptodateCheck = (options) => {
+  if (options.sourceBranch === options.targetBranch) {
+    return;
+  }
+
+  let spinner = ora(`Checking if ${options.sourceBranch} is uptodate with ${options.targetBranch}...`).start();
+
+  const res = shell.exec(`git rev-list "${options.sourceBranch}..${options.targetBranch}" --no-merges | wc -l`);
+  checkShellResponse(options, spinner, res);
+  if (parseInt(trim(res.stdout)) > 0) {
+    spinner.fail(
+      `Not all commits on ${options.targetBranch} are merged back into ${options.sourceBranch}.\n` +
+        `Please merge the following commits back into ${options.sourceBranch}:`
+    );
+    shell.exec(`git log --oneline --no-merges "${options.sourceBranch}..${options.targetBranch}"`, { silent: false });
+    dryRunOrNoDryRun(options, () => shell.exit(1));
+  }
+
+  spinner.succeed("Everything is up to date.");
+};
+
+const latestVersion = (options) => {
+  spinner = ora(`Getting lastest version...`).start();
+  let version = null;
+  shell.exec(`git fetch ${options.upstream} ${options.sourceBranch} refs/tags/*:refs/tags/* --prune`);
+  // Get tags, sort by date that tag was added and limit to last 50 for sanity
+  const tags = shell.exec(`git tag --sort=taggerdate | head -n 50`, { silent: true });
```

Comment:
Dit kan potentieel wel fout gaan wanneer je bv versie 1.3 hebt gemaakt maar nog niet releaset en een versie 1.2.1 wilt maken. Dan is de datum van 1.3 < 1.2.1.